spring-graphql/src/main/java/org/springframework/graphql/server/webmvc/GraphQlHttpHandler.java

```diff
@@ -102,6 +109,88 @@ public ServerResponse handleRequest(ServerRequest serverRequest) throws ServletE
 		return ServerResponse.async(responseMono);
 	}
 
+	public ServerResponse handleMultipartRequest(ServerRequest serverRequest) throws ServletException {
+		Optional<String> operation = serverRequest.param("operations");
+		Optional<String> mapParam = serverRequest.param("map");
+		Map<String, Object> inputQuery = readJson(operation, new TypeReference<>() {});
+		final Map<String, Object> queryVariables;
+		if (inputQuery.containsKey("variables")) {
+			queryVariables = (Map<String, Object>)inputQuery.get("variables");
+		} else {
+			queryVariables = new HashMap<>();
+		}
+		Map<String, Object> extensions = new HashMap<>();
+		if (inputQuery.containsKey("extensions")) {
+			extensions = (Map<String, Object>)inputQuery.get("extensions");
+		}
+
+		Map<String, MultipartFile> fileParams = getMultipartMap(serverRequest);
+		Map<String, List<String>> fileMapInput = readJson(mapParam, new TypeReference<>() {});
+		fileMapInput.forEach((String fileKey, List<String> objectPaths) -> {
+			MultipartFile file = fileParams.get(fileKey);
+			if (file != null) {
+				objectPaths.forEach((String objectPath) -> {
+					MultipartVariableMapper.mapVariable(
+						objectPath,
+						queryVariables,
+						file
+					);
+				});
+			}
+		});
+
+		String query = (String) inputQuery.get("query");
+		String opName = (String) inputQuery.get("operationName");
+
+		WebGraphQlRequest graphQlRequest = new WebGraphQlRequest(
+			serverRequest.uri(), serverRequest.headers().asHttpHeaders(),
+			query,
+			opName,
+			queryVariables,
+			extensions,
+			this.idGenerator.generateId().toString(), LocaleContextHolder.getLocale());
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("Executing: " + graphQlRequest);
+		}
+
+		Mono<ServerResponse> responseMono = this.graphQlHandler.handleRequest(graphQlRequest)
+			.map(response -> {
+				if (logger.isDebugEnabled()) {
+					logger.debug("Execution complete");
+				}
+				ServerResponse.BodyBuilder builder = ServerResponse.ok();
+				builder.headers(headers -> headers.putAll(response.getResponseHeaders()));
+				builder.contentType(selectResponseMediaType(serverRequest));
+				return builder.body(response.toMap());
+			});
+
+		return ServerResponse.async(responseMono);
+	}
+
+	private <T> T readJson(Optional<String> string, TypeReference<T> t) {
```

Comment:
@rstoyanchev Hi,
I am not sure if it possible to save backward compatible constructor in 1.0.x.
I need to parse strings somehow, so I need to add `ObjectMapper` (or `List<HttpMessageConverter<?>> messageConverters` as in `DefaultServerRequest`).

Reply:
Sorry for dummy question.
Seems I can remain existing constructor, and use new ObjectMapper there, and add a new one 
```
	/**
	 * Create a new instance.
	 * @param graphQlHandler common handler for GraphQL over HTTP requests
	 */
	public GraphQlHttpHandler(WebGraphQlHandler graphQlHandler) {
		Assert.notNull(graphQlHandler, "WebGraphQlHandler is required");
		this.graphQlHandler = graphQlHandler;
        this.objectMapper = new ObjectMapper();
	}

    /**
     * Create a new instance.
     * @param graphQlHandler common handler for GraphQL over HTTP requests
     * @param objectMapper ObjectMapper used for parsing form parts
     */
    public GraphQlHttpHandler(WebGraphQlHandler graphQlHandler, ObjectMapper objectMapper) {
        Assert.notNull(graphQlHandler, "WebGraphQlHandler is required");
        Assert.notNull(objectMapper, "ObjectMapper is required");
        this.graphQlHandler = graphQlHandler;
        this.objectMapper = objectMapper;
    }
```